manifests/fpm.pp

```diff
@@ -45,20 +45,20 @@
 #
 class php::fpm (
   $ensure               = $::php::ensure,
-  $service_ensure       = $::php::params::fpm_service_ensure,
-  $service_enable       = $::php::params::fpm_service_enable,
-  $service_name         = $::php::params::fpm_service_name,
-  $service_provider     = undef,
-  $package              = "${::php::package_prefix}${::php::params::fpm_package_suffix}",
-  $inifile              = $::php::params::fpm_inifile,
-  $settings             = {},
-  $global_pool_settings = {},
-  $pools                = { 'www' => {} },
-  $log_owner            = $::php::params::fpm_user,
-  $log_group            = $::php::params::fpm_group,
-) inherits ::php::params {
+  $service_ensure       = $::php::fpm_service_ensure,
+  $service_enable       = $::php::fpm_service_enable,
+  $service_name         = $::php::fpm_service_name,
+  $service_provider     = $::php::fpm_service_provider,
+  $package              = $::php::real_fpm_package,
+  $inifile              = $::php::fpm_inifile,
+  $settings             = $::php::real_settings,
+  $global_pool_settings = $::php::real_fpm_global_pool_settings,
+  $pools                = $::php::real_fpm_pools,
+  $log_owner            = $::php::log_owner,
+  $log_group            = $::php::log_group,
+) {
 
-  if $caller_module_name != $module_name {
+  if ! defined(Class['php']) {
     warning('php::fpm is private')
```

Comment:
Not sure the warning message is still matching the test. If you include the ```php``` class in your manifest, you can now use the ```php::fpm``` class which is then no more ```private``` (but I might be wrong)